packages/plugin-compiler/src/main/kotlin/io/realm/compiler/RealmModelSyntheticPropertiesGeneration.kt

```diff
@@ -335,23 +356,39 @@ class RealmModelSyntheticPropertiesGeneration(private val pluginContext: IrPlugi
                                             symbol = type.symbol
                                         )
                                     )
-                                    // Collection type
+                                    // Collection type: remember to specify it correctly here - the
+                                    // type of the contents itself is specified as "type" above!
+                                    val collectionTypeSymbol = when (value.collectionType) {
+                                        CollectionType.NONE -> PROPERTY_COLLECTION_TYPE_NONE
+                                        CollectionType.LIST -> PROPERTY_COLLECTION_TYPE_LIST
+                                        CollectionType.SET -> error("Sets not ready yet.")
```

Comment:
Maybe more expressive `Unsupported collection type for field ...` to lead the way for fixing thing when you get this error.